python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -121,33 +122,33 @@ def __init__(
             self._default_load_type = default_load_type
 
     def handle_output(self, context: OutputContext, obj: object) -> None:
-        table_slice = self._get_table_slice(context, context)
+        # If the output type is set to Nothing, handle_output will not be
+        # called. We still need to raise an error when the return value
+        # is None, but the typing type is not Nothing
+        if obj is None:
+            raise DagsterInvariantViolationError(
+                "Unexpected 'None' output value. If a 'None' value is intentional, set the output type to None.",
```

Comment:
should we be more explicit in the hint? as in using similar language as https://github.com/dagster-io/dagster/pull/17451/files like `add return type annotation '-> None'`